### PR TITLE
Moving balloc static memory buffer to DCCM memory

### PR DESCRIPTION
**Pull request by SidLeung**
balloc() is used by interrupt context code for memory allocation. The memory space is statically allocated at
initialization.  In order not to take up SRAM memory, this buffer is moved to the DCCM memory space.

### CODE REVIEW

**Comment by SidLeung** [comment]
@bigdinotech @eriknyquist Would you please check out this fix for moving the balloc() static memory buffer to the DCCM area?  Thanks.

**Comment by noelpaz** [comment]
The available memory is bigger now 9556. I also run the BAT and compilation tests. @SidLeung , do you want to reset the i2s example sketches back to the original?

**Comment by noelpaz** [comment]
I am testing BLE and will give update ASAP

**Comment by noelpaz** [comment]
Tested all the examples (just waiting for @russmcinnis to report out on Sensor Tag ) . Also tested backward compatibility

**Comment by SidLeung** [comment]
@calvinatintel Please merge.  @noelpaz and @russmcinnis tested this PR and it was okay with them.

File: system/libarc32_arduino101/framework/src/os/balloc.c [hunk 1]
```diff
@@ -10,6 +10,9 @@
 #include "os/os.h"
 #include "infra/log.h"
 
+extern void* dccm_memalign(uint16_t size);
```
Comment (eriknyquist) [inline system/libarc32_arduino101/framework/src/os/balloc.c]: Is it possible to include "dccm_alloc.h"?

File: system/libarc32_arduino101/framework/src/os/balloc.c [hunk 2]
```diff
@@ -98,23 +100,22 @@ T_POOL_DESC mpool[] =
 
 #else
 
+
```
Comment (eriknyquist) [inline system/libarc32_arduino101/framework/src/os/balloc.c]: Don't need the extra blank line here

File: system/libarc32_arduino101/framework/src/os/balloc.c [hunk 3]
```diff
@@ -98,23 +100,22 @@ T_POOL_DESC mpool[] =
...
-
 /** Pool descriptor definition */
 T_POOL_DESC mpool [] =
 {
 #define DECLARE_MEMORY_POOL(index, size, count) \
     { \
 /* T_POOL_DESC.track */ mblock_alloc_track_ ## index, \
-/* T_POOL_DESC.start */ (uint32_t)mblock_ ## index, \
-/* T_POOL_DESC.end */ (uint32_t)mblock_ ## index + count * size, \
+/* T_POOL_DESC.start */ (uint32_t)NULL, \
```
Comment (eriknyquist) [inline system/libarc32_arduino101/framework/src/os/balloc.c]: Why "(uint32_t)NULL", instead of just "0"?

**Comment by noelpaz** [comment]
@russmcinnis tested the sensor tag example and it looks we are good to go for a merge

File: system/libarc32_arduino101/framework/src/os/balloc.c (hunk 1 above)
Comment (SidLeung) [inline system/libarc32_arduino101/framework/src/os/balloc.c]: Add the cores/arduino path to the Makefile to use dccm_alloc.h instead.

File: system/libarc32_arduino101/framework/src/os/balloc.c (hunk 3 above)
Comment (SidLeung) [inline system/libarc32_arduino101/framework/src/os/balloc.c]: Yes, will do that.

**Comment by SidLeung** [comment]
@eriknyquist Fixed up your requests.  Please check it out and let @calvinatintel know.  Thanks.

**Comment by SidLeung** [comment]
@eriknyquist Make c_init.c to use the include file too.  Please check it out and inform Calvin.

**Pull request merged**